### ISSUE TITLE
Cholesky decomposition was not successful. Eigen::LLT failed with error code 1. Filling lower-triangular output with NaNs

### ISSUE THREAD

**Issue by L1021204735** [ZPZhou-lab/tfkan#17]
当我运行您的示例grid_extension_demo时，我能得到同样的结果。但当我自己编写代码时，发现程序在调用DenseKAN的extend_grid_from_samples方法后会抛出warning：tensorflow/core/kernels/linalg/cholesky_op.cc:56] Cholesky decomposition was not successful. Eigen::LLT failed with error code 1. Filling lower-triangular output with NaNs.请问这是为什么？您辛苦了！

以下是我的代码片段，其中train_model_lbfgs用的是您grid_extension_demo里的方法。

x = df_encoded.iloc[:, 4:6].to_numpy()[:, :].astype(np.float64)
x_min = x.min(axis=0)
x_max = x.max(axis=0)
x_normalized = (x - x_min) / (x_max - x_min)
x_normalized = 2 * x_normalized - 1

y = df_encoded[['grade']].to_numpy().astype(np.float64)

x_train, x_test, y_train, y_test = train_test_split(x_normalized, y, test_size=0.3)
y_train, y_test = y_train.reshape(-1, 1), y_test.reshape(-1, 1)

kan = tf.keras.models.Sequential([
    DenseKAN(5, grid_size=5, dtype=tf.float64),
    DenseKAN(1, grid_size=5, dtype=tf.float64)
])
kan.build(input_shape=(None, 2))
kan.summary()

extend_epoch = 300
train_mse, valid_mse = train_model_lbfgs(
    kan, x_train, y_train, x_test, y_test, epoches=1800, verbose=100, extend_epoch=extend_epoch)
